apps/client/src/AppRouter.tsx

```diff
@@ -42,9 +42,27 @@ export default function AppRouter() {
     if (!data) return;
 
     for (const d of data) {
-      if (`/${d.alias}` === location.pathname && d.enabled) {
-        navigate(`/${d.pathAndParams}`);
-        break;
+      if (location.search.indexOf('alias=') > -1) {
+        // if the alias fits the alias on this page, if the URL is diferent, we redirect user to the new URL
+        let aliasOnPage = location.search.substring(location.search.indexOf('alias=') + 6);
+        if (aliasOnPage.indexOf('&') > -1) {
+          // take care of scenario where there is an & after the alias
+          aliasOnPage = aliasOnPage.substring(0, aliasOnPage.indexOf('&'));
+        }
+        // if we have the same alias and its enabled and its not empty
+        if (d.alias !== '' && d.enabled && d.alias === aliasOnPage) {
+          let fullLocationPath = location.pathname + location.search;
+          // we need to remove the &alias=ALIAS_NAME or ?alias=ALIAS_NAME at the end of the URL
+          fullLocationPath = fullLocationPath.replaceAll(`alias=${d.alias}`, '');
+          fullLocationPath = fullLocationPath.substring(0, fullLocationPath.length - 1);
+          // if the location path is different
+          if (d.pathAndParams !== fullLocationPath) {
+            //console.log('found a match, same alias but different route so I will redirect to: ', d.pathAndParams);
+            const connector = d.pathAndParams.includes('?') ? '&' : '?';
+            navigate(`${d.pathAndParams}${connector}alias=${d.alias}`);
```

Comment:
Hmm, I wonder if there is a better way of doing this.
The react router documentation is not particular specific on this, but this thread seems to hint to the ability of being able to pass the values programmatically 
https://github.com/remix-run/react-router/issues/7743